### ISSUE TITLE
Manual re-install required after OS update

### ISSUE THREAD

**Issue by gelraen** [tekezo/Karabiner-Elements#1930]
Updated to Mac OS X 10.14.6 today and Karabiner stopped working. Pressing restart button in the Misc tab didn't help at all, as well as ignoring and unignoring the keyboard in the Devices tab. Also, there's no indication of any issues, other then things simply not working, e.g., my caps lock doing caps lock stuff, instead of being remapped to F19.

I also remember the same thing happening after the previous OS update. At this point even a simple popup asking for re-install would be a huge step forward in avoiding spending a lot of time debugging the issue.

Log after re-install: [Karabiner reinstall log 2019-07-31.log](https://github.com/tekezo/Karabiner-Elements/files/3451713/Karabiner.reinstall.log.2019-07-31.log)

**Comment by flowen** [comment]
I had the same issue after installing 10.14.6

Also suddenly a rule I installed to change capslock to shift+cmd+control was now changed to shift+cmd+alt+control. 

Had to rebind some keys in BetterTouchTool because of this. Not huge, but didn't expect at all

**Comment by rprimus** [comment]
Fri Aug  2 13:28:49 BST 2019

Installed the `Supplementary Update` today and had the same problem - `Karabiner` was effectively not working.

I restarted Karabiner, but no effect.

Quit Karabiner and came here looking to see if this had been reported.

After starting to reply here, I wanted to show output of `ps axuw | ag -i
Karabiner`.  Noticed Karabiner was not running and started it.

Surprisingly - (Karabiner was **down** for 17 mins) - all is well now.  There were no errors in the Karabiner log.

I had similar errors to @gelraen - when Karabiner was not working:

grep 'd errors from @gelraen provided log:

```
40:[2019-07-31 12:22:34.380] [error] [grabber] console_user_server_client error: Connection reset by peer
51:[2019-07-31 12:22:39.936] [error] [observer] grabber_client error: Connection reset by peer
101:[2019-07-31 12:38:18.274] [error] [observer] grabber_client error: Connection reset by peer
164:[2019-07-31 12:59:02.128] [error] [session_monitor] session_monitor_receiver_client error: Destination address required
165:[2019-07-31 12:59:02.128] [error] [session_monitor] session_monitor_receiver_client error: Destination address required
166:[2019-07-31 12:59:02.128] [error] [session_monitor] session_monitor_receiver_client error: Destination address required
```

HTH.

**Comment by jgclark** [comment]
CapsLock behaviour in my Hyper key setup broke on MacMini some days ago -- probably related to upgrade to 10.14.6.  Fresh installs of 12.5.0 and 12.5.7 don't help. 
But 12.5.7 on 10.14.6 on MacBook Air work fine. So it made me think it's the keyboard on my MacMini, but swapping in another keyboard has the same issue.
Looking in EventViewer on MacMini with no modifications running, it seems that I get the expected caps_lock key_down event, but not the expect key_up event. Whereas on MacBook I get both key_down and key_up events.
No errors in log.
Weird.

**Comment by cksachdev** [comment]
Tab key and Caps lock key stopped working after Mac OS X 10.14.6 update. This seems to affect only my External Apple Keyboard(wired). The one on the laptop seems to work fine. Checking in event viewer pressing capslock and tab key(tilde and escape key too) don't capture anything. Tried following:
1. Restart the system
2. Reinstall Karabiner
3. Check karabiner config to ensure it matches my previous config
4.  Quit Karabiner and pgrep karabiner | xargs sudo kill -9
5. Restart the sytem
None of these steps seem to work for me. Sharing my Karabiner config. Is there any other step which I should try to make it work?

**Comment by ns-mkusper** [comment]
In my case restarting karabiner, my OS, etc and even uninstalling did not work to get karabiner functional again. I eventually found that, although the popup was telling me simply to restart MacOS (or disable the driver which I did many times) I actually simply needed to allow the driver in the MacOS security settings. I know there is some functionality in Karabiner to notify you of this status so I'm not sure why it was not triggered for me following an update to Sonora.
